bundle/src/main/java/com/adobe/acs/commons/properties/filter/CapturingResponseWrapper.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2020 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+package com.adobe.acs.commons.properties.filter;
+
+import javax.servlet.ServletOutputStream;
+import javax.servlet.WriteListener;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpServletResponseWrapper;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+
+/**
+ * Response Wrapper used to allow for full output of the current response data so that it can be manipulated
+ * and rewritten.
+ */
+public class CapturingResponseWrapper extends HttpServletResponseWrapper {
+
+    private final ByteArrayOutputStream capture;
+    private ServletOutputStream output;
+    private PrintWriter writer;
+
+    public CapturingResponseWrapper(HttpServletResponse response) {
+        super(response);
+        capture = new ByteArrayOutputStream(response.getBufferSize());
+    }
+
+    @Override
+    public ServletOutputStream getOutputStream() {
+        if (writer != null) {
+            throw new IllegalStateException(
+                    "getWriter() has already been called on this response.");
+        }
+
+        if (output == null) {
+            output = new ServletOutputStream() {
+                @Override
+                public void write(int b) throws IOException {
+                    capture.write(b);
+                }
+
+                @Override
+                public void flush() throws IOException {
+                    capture.flush();
+                }
+
+                @Override
+                public void close() throws IOException {
+                    capture.close();
+                }
+
+                @Override
+                public boolean isReady() {
+                    return false;
+                }
+
+                @Override
+                public void setWriteListener(WriteListener writeListener) {
+                    // do nothing
+                }
+            };
+        }
+
+        return output;
+    }
+
+    @Override
+    public PrintWriter getWriter() throws IOException {
+        if (output != null) {
```

Comment:
Any reason we dont use `response.getWriter()`?

Reply:
I think this is overwritten so that we can actually capture what's being written and interact with it instead of just outputting the info

Reply:
ah k